src/apis/dataset/DataCatalogLODHandler.py

```diff
@@ -16,14 +18,44 @@ class DataCatalogLODHandler:
     def __init__(self, config):
         self.config = config
         self.logger = logging.getLogger(config["LOG_NAME"])
-        self._data_catalog = self._parse_catalog_file(config["DATA_CATALOG_FILE"])
-
-    # TODO: implement caching suitable for this function
-    def _parse_catalog_file(self, path: str) -> Graph:
-        """Parse catalog file (turtle) and return graph of results'"""
-        self.logger.info(f"Loading data catalogue from '{path}'")
-        graph = Graph().parse(path, format=MimeType.TURTLE.value)
-        return graph
+        self.cache = config["GLOBAL_CACHE"]
+        sparql_endpoint = self.config.get("SPARQL_ENDPOINT")
+        self._data_catalog = self._get_data_catalog_from_store(sparql_endpoint)
+
+    def _get_data_catalog_from_store(
+        self, sparql_endpoint: str, cache_key: str = "data_catalog", minutes: int = 0
+    ) -> Graph:
+        """Get data catalog triples from the rdf store and return graph. Simple caching
+        is enabled as well as a simple expiration mechanism for the cache."""
+        cache_key_expiration = f"{cache_key}_expiration"
+        if cache_key_expiration in self.cache:
+            if datetime.utcnow() >= self.cache[cache_key_expiration]:
+                self.logger.debug(
+                    f"The cache for {cache_key} is expired. Emptying cache."
+                )
+                del self.cache[cache_key]
+                del self.cache[cache_key_expiration]
+
+        if cache_key in self.cache:
+            self.logger.debug(f"GOT THE {cache_key} FROM CACHE")
+            return self.cache[cache_key]
+        else:
+            self.logger.debug(f"NO {cache_key} FOUND IN CACHE")
+            self.logger.info(f"Getting data catalog triples from '{sparql_endpoint}'")
+            construct_query = (
+                "CONSTRUCT { ?sub ?pred ?obj } WHERE { "
+                "GRAPH <http://data.rdlabs.beeldengeluid.nl/datacatalog/> { ?sub ?pred ?obj } }"
```

Comment:
Yes that should be configurable, but since this implies changes on the server I was reluctant to insert that immediately. If we stick with the same named graph for the data catalog it should not be a problem for now.

I will also remove the cache functionality for the datacatalog